enterprise/internal/campaigns/reconciler.go

```diff
@@ -807,6 +884,7 @@ func CompareChangesetSpecs(previous, current *campaigns.ChangesetSpec) (*changes
 type changesetSpecDelta struct {
 	titleChanged         bool
 	bodyChanged          bool
+	draftChanged         bool
```

Comment:
Can't we build it like this and use methods to determine what we need to do?

```suggestion
	publishedChanged         bool
```